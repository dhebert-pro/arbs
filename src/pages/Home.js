import React from "react";
import Footer from "components/Footer";
import Homepage from "components/Homepage";
import Header from "components/Header";
import App from "components/App";
import glamorous from "glamorous";

const titleStyles = {
  color: "red"
};

const Title = glamorous.div(titleStyles);

const Page = ({ posts }) => (
  <App>
    <div>
      <Header />
      <Title>Ceci est un titre</Title>
      <Homepage datas={posts} />
      <Footer />
    </div>
  </App>
);

export default Page;
