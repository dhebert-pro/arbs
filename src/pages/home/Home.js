import React from "react";
import Footer from "components/Footer";
import Homepage from "components/Homepage";
import Header from "components/Header";
import App from "components/App";

const Page = ({ datas }) => (
  <App className="App Homepage">
    <div>
      <Header />
      <Homepage datas={datas} />
      <Footer />
    </div>
  </App>
);

export default Page;
