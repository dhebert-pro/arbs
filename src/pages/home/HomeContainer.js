import axios from "axios";

import Home from "./Home";

Home.getInitialProps = () => {
  return axios
    .get("https://jsonplaceholder.typicode.com/posts")
    .then(result => {
      return { datas: result.data };
    });
};

export default Home;
