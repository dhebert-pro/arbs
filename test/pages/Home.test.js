/*global jest expect test describe:true */
/*eslint no-undef: "error"*/
import React from "react";
import renderer from "react-test-renderer";
import Home from "pages/Home";
import HomeContainer from "pages";

jest.mock("components/App", () => ({ children }) => (
  <div name="App">{children}</div>
));
jest.mock("components/Header", () => () => <div name="Header" />);
jest.mock("components/Footer", () => () => <div name="Footer" />);
jest.mock("components/Homepage", () => ({ datas }) => (
  <div name="Homepage">{JSON.stringify(datas)}</div>
));

describe("Home Page", () => {
  test("should be rendered", done => {
    const component = renderer.create(<Home posts="TEST" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});

describe("Home Container", () => {
  test("should be rendered", done => {
    const component = renderer.create(<HomeContainer />);
    expect(component).toMatchSnapshot();
    done();
  });
  test("should get posts", done => {
    const configureStore = require.requireActual("store/configureStore");
    configureStore.fetchPosts = jest.fn(() => () => "TEST");
    HomeContainer.getInitialProps().then(result => {
      expect(result.initialProps).toEqual("TEST");
      done();
    });
  });
});
