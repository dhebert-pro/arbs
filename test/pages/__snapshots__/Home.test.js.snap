// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Container should be rendered 1`] = `
<div
  name="App"
>
  <div>
    <div
      name="Header"
    />
    <div
      className="css-1ezp9xe"
    >
      Ceci est un titre
    </div>
    <div
      name="Homepage"
    />
    <div
      name="Footer"
    />
  </div>
</div>
`;

exports[`Home Page should be rendered 1`] = `
<div
  name="App"
>
  <div>
    <div
      name="Header"
    />
    <div
      className="css-1ezp9xe"
    >
      Ceci est un titre
    </div>
    <div
      name="Homepage"
    >
      "TEST"
    </div>
    <div
      name="Footer"
    />
  </div>
</div>
`;
